### ISSUE TITLE
Plain text editing of markdown files

### ISSUE THREAD

**Issue by noe-spi** [nextcloud/text#439]
When installing Nextcloud Markdown Editor ([Link](https://github.com/icewind1991/files_markdown)) and trying to open a xx.md or xx.mardown file the text editor opens.
 
**To Reproduce**
1. install Nextcloud Markdown Editor
2. Create a new file (or use an existing one) that shall be a markdown file (xx.md / xx.markdown) 
3. Click on the newly created file to open
4. See error --> Text Editor will open and nor the Markdown Editor

**Expected behavior**
Markdown Editor shall open

**Client details:**
 - OS: on each OS
 - Browser: firefox, chrome, opera
 - Device: Desktop

**Text app version:** 
1.1.1

**Operating system:**
Linux (Ubuntu 18.04) running as docker image
**Web server:**
Apache
**Database:**
MySQL
**PHP version:**
7.2
**Nextcloud version:** (see Nextcloud admin page)
17.0.1
</details>

**Comment by rollanders** [comment]
Another problem - I deactivated Text, and had still Markdown editor activated, but now the md-file is not opened but the download windows opens.

**Comment by ufgum** [comment]
Same problem here. The PicoCMS needs the yaml-header, but text destroys it.

**Comment by noe-spi** [comment]
see [mitigation](https://github.com/icewind1991/files_markdown/issues/136#issuecomment-560134316 ) for a solution wrt markdown editor.
But the main problem is here I think: [Nextclouds decision to focus on the new Text app and discontinue the development of the old plain text editor](https://github.com/nextcloud/files_texteditor/issues/185)

**Comment by juliushaertl** [comment]
@jancborchardt Not sure what we should do here from a UX perspective. Should the markdown editor take precedence over the text app when installed?

**Comment by E1k3** [comment]
@juliushaertl As long as the markdown editor is even working (ie when the old text editor app is installed), it is possible to open a md file in the markdown editor via the context menu. While this is not perfect, it is something I could accept. In the long run, I would prefer a system, where you can select preferred apps for certain filetypes.

The real problem right now is, that the old TextEditor App was taken off the store and without it, the markdown editor does not work at all.

**Comment by blvb** [comment]
It is necessary to change min-version to 17 in apps/files_texteditor/appinfo/info.xml
As follows:
     <dependencies>
        <nextcloud min-version="17" max-version="18" />
    </dependencies>
Seems to work properly now for NC 17.0.1

**Comment by jancborchardt** [comment]
So this doesn’t seem to be 1 big issue but several smaller ones which need to be treated separately and can be fixed separately too:

------------------------

> I want the Markdown editor to edit my index.md files for PicoCMS. Especially the yaml-header. The Text-app destroys the header.

@rollanders @ufgum this of course is an issue which should ideally be fixed in the Text app. Could you open a separate issue about this? → https://github.com/nextcloud/text/issues/439

--------------------

@noe-spi @E1k3 @inpos @lanxu @blvb what are the actual issues you encounter with Nextcloud Text as Markdown editor which could possibly be fixed? :) Could you open separate issues about those too? → https://github.com/nextcloud/text/issues/new

----------------

> The real problem right now is, that the old TextEditor App was taken off the store and without it, the markdown editor does not work at all.

@E1k3 this is an issue that should be fixed in the Markdown app → https://github.com/icewind1991/files_markdown cc @icewind1991 


-------------


That leaves the question to decide:
**When both Nextcloud Text (default) and the Markdown app are installed, what should open if you open a Markdown file?**

- **Text:** Is the default, is collaborative.
- **Markdown app:** Is not collaborative, but when you specifically installed it you could argue that you would rather use this one? But why not disable the Text app then?

@juliushaertl Until these questions are resolved I’d prefer to keep the current state. But understand this is not set in stone – there’s just many small steps to take instead of one big fix. :)

**Comment by E1k3** [comment]
> @E1k3 this is an issue that should be fixed in the Markdown app → https://github.com/icewind1991/files_markdown cc @icewind1991

As far as I understand, the Markdown app was just an extension of the old TextEditor app maintained by the Nextcloud team. While I agree, that this is not a problem related to the new Text app (so not one to discuss here, but in the TextEditor repo https://github.com/nextcloud/files_texteditor/issues/185), I wouldn't call it a bug/problem of the Markdown app either. The TextEditor app vanishing from the store should not have to be fixed by the extensions that were developed for it.

> @noe-spi @E1k3 @inpos @lanxu @blvb what are the actual issues you encounter with Nextcloud Text as Markdown editor which could possibly be fixed? :) Could you open separate issues about those too?

The only issue I have with the Text app, is that is cannot replace a raw + preview editor, but that is not something a WYSIWYG editor will ever do (or is supposed to do), so this doesn't fulfill the condition of "could possibly be fixed".

**Comment by shankari** [comment]
I would also like the ability to have a non-rich text editor. I understand that most people want rich text, and the collaborative features are cool.

But sometimes, I just want to write notes to myself, and I already know markdown syntax, so I want to write in plain text so I don't have to take my hands off the keyboard.

The rich text editor has some quirks - e.g. if I type the `####` and then `<space>`, then the hashes disappear, and if I want to go to a higher or lower heading level, then I have to use the dropdown menu. One of the benefits of markdown is that you can type it all as raw text without using the mouse and I would like to keep using that benefit.

I loved the previous plain text editor + preview on the side option, and would really appreciate the  ability to configure my running instance to have it as an option.

**Comment by noe-spi** [comment]
> So this doesn’t seem to be 1 big issue but several smaller ones which need to be treated separately and can be fixed separately too:
> 
> > I want the Markdown editor to edit my index.md files for PicoCMS. Especially the yaml-header. The Text-app destroys the header.
> 
> @rollanders @ufgum this of course is an issue which should ideally be fixed in the Text app. Could you open a separate issue about this? → #439
> 
> @noe-spi @E1k3 @inpos @lanxu @blvb what are the actual issues you encounter with Nextcloud Text as Markdown editor which could possibly be fixed? :) Could you open separate issues about those too? → https://github.com/nextcloud/text/issues/new
> 
> > The real problem right now is, that the old TextEditor App was taken off the store and without it, the markdown editor does not work at all.
> 
> @E1k3 this is an issue that should be fixed in the Markdown app → https://github.com/icewind1991/files_markdown cc @icewind1991
> 
> That leaves the question to decide:
> **When both Nextcloud Text (default) and the Markdown app are installed, what should open if you open a Markdown file?**
> 
>     * **Text:** Is the default, is collaborative.
> 
>     * **Markdown app:** Is not collaborative, but when you specifically installed it you could argue that you would rather use this one? But why not disable the Text app then?
> 
> 
> @juliushaertl Until these questions are resolved I’d prefer to keep the current state. But understand this is not set in stone – there’s just many small steps to take instead of one big fix. :)

... sorry to respond as late - it was not possible to me to react eralier.
Let me summarize from my point of view:

1) MarkDown Editor need the old TextEditor - not maintained anymore by Nextcloud 

2) the new Text APP do not support the needs by Markdown Editor. Here I would ask if the Text App is the official replacement for TextEditor by Nextcloud? -> I think so

3) Text APP and TextEditor do not coexist in one Nextcloud Instance

4) Text APP opens automatically *.md / *.markdown files (even if MarkdownEditor &&-|| TextEditor are installed)

==> Conclusio:
I agree that this is not an issue in one direction (neither MarkDown nor Text APP), but:

- as MarkdownEditor builds on TextEditor (need TextEditor) why did the two groups do not know from each other and their dependencies? Especially if there is the decision to replace TextEditor with Text APP.

- as the Text APP is a kind of "simple" WHSIWYG Editor - why does this APP automatically open *.md/*.markdown files and supersede other possible coexisting Editors (at least that is what I experienced)??? For me it is not comprehensible why Text APP open files with well  known endings that - for sure - do not belong to Text APP and overrule with that coexisting editors? 
And THIS was my initial question (as the headline of this issue express)

For me it is not clear whom to talk to. On the one hand I would like to use the MarkdownEditor further on but this would mean that both groups are talking to each other.
Especially I would be interested in what the developers of MarkdownEditor are guessing as they have that strong dependency to the old TextEditor (which is not supported any more) - are they planning to implement the Text-Engine on their own? Do they talk to Text APP developpers about their need of a text app that is capable to provide raw&preview editor?
And I would be interested in the reason why the Text APP developers implement the Text APP the way that it overrules other editors and opens files with file extensions that surely do not belong to them.

**Comment by noe-spi** [comment]
one question to the round ( and thanks for your engagement - especially to @E1k3 who helped me with his work around  ;-)  : 
  shall  I close this issue or leave it open for further discussions?

**Comment by E1k3** [comment]
> 4. Text APP opens automatically *.md / *.markdown files (even if MarkdownEditor &&-|| TextEditor are installed)

> -  as the Text APP is a kind of "simple" WHSIWYG Editor - why does this APP automatically open .md/.markdown files and supersede other possible coexisting Editors (at least that is what I experienced)??? For me it is not comprehensible why Text APP open files with well known endings that - for sure - do not belong to Text APP and overrule with that coexisting editors?
And THIS was my initial question (as the headline of this issue 

Even though the new Text app is a WYSIWYG editor, all of the formatting is done via markdown, so the Text app saves files as `.md` and is the default app to open `.md` files as well.

According to the [README.md](https://github.com/nextcloud/text), this is done to enable other editors to edit your documents via markdown as a universal format. Especially on desktops where you want to edit the files in a native text editor instead of the web view, this makes a lot of sense.

As far as I know, it's still possible to edit `.md` files using another editor plugin/app while "Text" is enabled, using the context menu (right click the file, choose edit with ...). You could test that, by enabling both Text and files_texteditor (the old text editor). I would prefer if there was an option to chose the default app for certain file types though.

**Comment by E1k3** [comment]
@shankari wrote
> The rich text editor has some quirks - e.g. if I type the #### and then <space>, then the hashes disappear, and if I want to go to a higher or lower heading level, then I have to use the dropdown menu. One of the benefits of markdown is that you can type it all as raw text without using the mouse and I would like to keep using that benefit.

I agree, that this is counter intuitive, but the Text app is really a markdown editor. It just renders the text differently. So when your `#` characters disappear, they are still there, changing the heading level of the following text and you can still remove heading levels by pressing backspace. They are just not rendered.

This is really weird because, for example, you can only guess the current heading level by the font size.

**Comment by shankari** [comment]
@E1k3 

> So when your # characters disappear, they are still there, changing the heading level of the following text and you can still remove heading levels by pressing backspace. 

I know that they are still there, but it is jarring and unintutitive. I guess my real issue is that I don't like WSYWIG editors. I prefer to edit the raw text. I understand that I am in a minority, but given that I am running my own instance of nextcloud, I would really be able to configure it in a way that I prefer.

**Comment by E1k3** [comment]
@shankari  I definitely agree that it is unintuitive and I am not a fan of it either. I will keep using my old [vimwiki](https://github.com/vimwiki/vimwiki) setup without web editor until there is a supported alternative for nextcloud.

**Comment by jancborchardt** [comment]
@juliushaertl following up to my previous comment at https://github.com/nextcloud/text/issues/439#issuecomment-563135822

Does it make sense to add a "Plain text" toggle to the Text app?
- Is that supported by the editor?
- Would it still be collaborative then?
- Can we do this on a per-document or -filetype basis and remember that? Also e.g. by default .html or .css files would of course always be opened as plain text.

**Comment by christianlupus** [comment]
I have also the problem, that I need to change the raw markdown text from time to time. An example is if I let some snippet generate from other sources (browser extension), I need to download the md file, modify using local editor and then reupload. Here a remote raw editing feature would be very helpful.

**Comment by juliushaertl** [comment]
> this of course is an issue which should ideally be fixed in the Text app. Could you open a separate issue about this? → #439

Yes, that is a separate issue.

>> The real problem right now is, that the old TextEditor App was taken off the store and without it, the markdown editor does not work at all.
>  this is an issue that should be fixed in the Markdown app

We are currently preparing an appstore release of the old text editor so it will work again once that is done https://github.com/nextcloud/files_texteditor/pull/192

> When both Nextcloud Text (default) and the Markdown app are installed, what should open if you open a Markdown file?

> Text: Is the default, is collaborative.
> Markdown app: Is not collaborative, but when you specifically installed it you could argue that you would rather use this one? But why not disable the Text app then?

Yes, would make sense I guess to use the markdown editor by default if enabled as this is a conscious decision by the admin. But I need to have a look how the current file actions decide on the precedence.

> Does it make sense to add a "Plain text" toggle to the Text app?
> Is that supported by the editor? >Would it still be collaborative then?

It would make sense, however I expect the implementation to become quite complex as due to the way collaborative editing works, we cannot have different users open the file in different modes (plain/rich) at the same time. Let's say userA opens the file in rich editing mode and then userB opens it in plain editing mode, we need to make sure to end the session of userA and switch it to a plain editing session. Or maybe show a message to userB that this is not possible if multiple users are joined a document.

> Can we do this on a per-document or -filetype basis and remember that?
Per document, es with the above limitations. Filetype based doesn't make much sense as it is only about markdown files. Per user also would come with the above described limitations.

> Also e.g. by default .html or .css files would of course always be opened as plain text.

This is already the case.

**Comment by jancborchardt** [comment]
@juliushaertl if we already open html and css files in plain text, it seems a lot of the issues of people here would be fixed simply by adding more filetypes to that list? Like .yml for example – or if that's done already then which else?

**Comment by juliushaertl** [comment]
> if we already open html and css files in plain text, it seems a lot of the issues of people here would be fixed simply by adding more filetypes to that list? Like .yml for example – or if that's done already then which else?

Aparently css/html were missing in the list, since the viewer doesn't support wildcards. So https://github.com/nextcloud/text/pull/509 will add more mime types as a first step.

**Comment by nikkilocke** [comment]
> due to the way collaborative editing works, we cannot have different users open the file in different modes (plain/rich) at the same time

Can you elaborate? It would be nice to fix this!

**Comment by WarpinWolf** [comment]
> As far as I know, it's still possible to edit `.md` files using another editor plugin/app while "Text" is enabled, using the context menu (right click the file, choose edit with ...).

Sorry - but how would you be able to do this - I do not see "edit with"...
![image](https://user-images.githubusercontent.com/15440918/72422322-81baca80-3782-11ea-970c-33dee9a4ed1f.png)

**Comment by silverhook** [comment]
To add my two cents, I would also prefer having a split view option with the source side-by-side with the preview.

The issue I am currently running into is that I have a lot MarkDown files with metadata in them – either a YAML block for [Pandoc][] or simpler tag-value for [Pelican][] – and I found the new rich text editor eat the metadata block. Luckily I could recover from an earlier version, but that should not happen. A(n annotated) plain-text mode would get rid of that issue.

[pandoc]: https://pandoc.org/
[pelican]: https://blog.getpelican.com/

**Comment by zerknorscht** [comment]
I would really love to see it like this:
having a "show source code" button (disabled by default) to edit raw and still having the wysiwyg-functionality on what was the preview-pane...
if the wysiwyg-pane would respect the code-view-pane an stops fu**ing up the formating (e.g. yaml headers, e.g. by pico-cms in _meta.md) I would be so happy...
In fact, I would like to throw a few bucks on it on Bountysource, If I could (don't know how to open bountys) 
But I drew a nice picture for you:
![neweditor_mockup](https://user-images.githubusercontent.com/54960972/76022282-1b069180-5f27-11ea-8a33-5e696f9d4ff2.png)

**Comment by aschoerg** [comment]
I also heavily vote for this issue. 

The Editor as-is is confusing if you know markdown, e.g. why are images so small? I searched 5min for the render view to get those images to scale until I found this issue...

Splitting into source and rendered view or even switching would help improving so much! e.g. GitHub/GitLab's way of using the two tabs "Write" and "Preview" when writing this issue.

I don't know the code base but **I suggest the .md files are opened in rendered mode and with some button you can enter split-screen to edit the file.**

**Comment by nikkilocke** [comment]
I just right click on the file, and choose "Edit in plain text editor". No problem.

**Comment by ifuchs** [comment]
@nikkilocke What apps to you have enabled? (Plain text Editor, Text, Markdown Editor, ...?) I am not able to edit any of the code type files regardless of which one or which combinmation of htose apps I have enabled. Also, there are no new edit menu items on a right click on my system.

**Comment by nikkilocke** [comment]
Ah - didn't realise it was an app - thought the developers had listened to this thread. I appear to have something called "Plain text editor".

**Comment by ifuchs** [comment]
Plain text editor is an app but unless you modify the app's xml info it will not run on v18. What version of NC are you running?

**Comment by juliushaertl** [comment]
> Plain text editor is an app but unless you modify the app's xml info it will not run on v18. What version of NC are you running?

It will as it is now released to the app store.

As a general comment on both split view as well as plain text markdown editing let me quote myself here again: https://github.com/nextcloud/text/issues/123#issuecomment-508723049

> ... unfortunately I don't see a way to bring this together with the collaborative editing. Prosemirror uses a document model that in our case either supports plaintext or rich editing, but not both at the same time. We have proper plaintext support for non-markdown files with #129 and rich editing for markdown files, which should cover most of the use cases. The markdown editor will still continue to work like before and can be used besides the text app.

> Since we aimed for a rich editor this is out of scope for the text app.

So closing this as a won't fix.

**Comment by rollanders** [comment]
No need any longer.  The Plain text editor and Markdown editor now does what I want after their updates.

**Comment by ifuchs** [comment]
The problem that I still have with the editor is that it adds carriage
returns to the end of text files that only have new lines. This in turn
causes problems down the line. I have tried changing the parameter in the
ace editor to respect line endings but so far I haven’t gotten that to work.

**Comment by mlncn** [comment]
@nicolad  and @rollanders could you confirm that you can add links in markdown syntax - `[links](https://example.com)` - and then share a preview link to the document that has a formatted link?

I'd like to confirm this works before asking [May First Movement Technology](https://mayfirst.org) (where we use a shared NextCloud) to add the [Plain Text Editor app](https://apps.nextcloud.com/apps/files_texteditor) to the instance.  Thanks greatly!

**Comment by rollanders** [comment]
I dont understand exactly what you want to doHere I did a public page with a link to an internal asset and a second link to an external site. 
Welcome | testpublic

| 
| 
|  | 
Welcome | testpublic

Pico is a stupidly simple, blazing fast, flat file CMS.
 |

 |

 |






Is this what you want? 



    Den måndag 4 maj 2020 23:40:46 CEST, Benjamin Melançon <notifications@github.com> skrev:  
 
 


@nicolad and @rollanders could you confirm that you can add links in markdown syntax - [links](https://example.com) - and then share a preview link to the document that has a formatted link?

I'd like to confirm this works before asking May First Movement Technology (where we use a shared NextCloud) to add the Plain Text Editor app to the instance. Thanks greatly!

**Comment by aschoerg** [comment]
@mlncn 

I can confirm that links work:
![image](https://user-images.githubusercontent.com/6042536/81039165-85e13400-8ea8-11ea-87d5-71a201e48f47.png)

The first link is pointing to https://google.com as expected. The second one is somehow useless, since it references relatively, but Nextcloud paths are different: Either you use the webdav path (`https://{domain}/remote.php/webdav/{path} which I have not yet found out how to copy since right click opens the nextcloud context menu) or create the internal link (something like `https://{domain}/f/123`), but this does not open/download the file but opens the directory and marks the file.

Maybe someone has an idea how to easily e.g. use an url for an image? (But can also be ignored since this issue has nothing to do with that :wink:)

**Comment by rollanders** [comment]
This is the content of my index.md file for the link i posted. I do not understand what you mean with using the WebDav path. You just follow the manual and use the relative path to the assets folder. 

---
Title: Welcome
Description: Pico is a stupidly simple, blazing fast, flat file CMS.
---
## My public test site
with a [link to internal assets](%assets_url%/Advenstljus.jpg)
and a second link to an [external site](https://framatalk.org/accueil/en/)

**Comment by aschoerg** [comment]
Hey cool, thank you for telling me that there would be a way! :+1: Already saw you mentioning [here](https://help.nextcloud.com/t/assets-url-keeps-changing-to-25asset-url-25/79664)

But basically this is just a url-encode error. But I am not familiar on which side...

**Comment by rollanders** [comment]
I have not seen this before. It does not seem to be a plain-text issue. It does not matter if I use an external editor. I need to do some checking.

**Comment by rollanders** [comment]
I cannot link directly to a photo apparently
Check it out now

![image](https://user-images.githubusercontent.com/36486009/81044171-15d8ab00-8eb4-11ea-856b-8e8222402795.png)

**Comment by aschoerg** [comment]
Just want to inform you, that images work as intendend when using external URLs or relative paths :+1:. 

Have not yet tried an absolute path (but most likely I will not use this because I then cannot use it cross-platform in browser, mobile and PC)

**Comment by shankari** [comment]
@rollanders @juliushaertl 

wrt

>  The markdown editor will still continue to work like before and can be used besides the text app.

> No need any longer. The Plain text editor and Markdown editor now does what I want after their updates.

I am not able to open `.md` files using the Markdown Editor.

When the Text app is enabled, I see a menu item for "Edit in plain text editor", but no menu item for "Edit in markdown editor".

<img width="310" alt="Screen Shot 2020-06-20 at 9 53 42 AM" src="https://user-images.githubusercontent.com/2423263/85207248-48136000-b2dc-11ea-95c6-637ed56e12bc.png">

When the Text app is disabled, there are no menu items, and the file opens in the plain text editor.

Are there specific instructions on how to open `.md` files with the markdown editor?

Versions:
- nextcloud: 17.0.1
- markdown editor: 2.3.0
- plain text editor: 2.11.0
- Text app: 1.1.1

<img width="1050" alt="Screen Shot 2020-06-20 at 10 00 00 AM" src="https://user-images.githubusercontent.com/2423263/85207320-d2f45a80-b2dc-11ea-8ae1-1369c6ea3443.png">

Happy to upload logs if necessary.

**Comment by shankari** [comment]
Answering my own question for people who find this bug in the future.
- upgrade to nextcloud 19
- ensure that the Markdown Editor is enabled and is at version 2.3.0
- disable the Text app
- now, if you click on any `.md` file file, it opens in the markdown editor which offers a split-screen with edit and preview panes

Note that this option **does not** appear in the action menu. you have to **disable** the Text app for the Markdown Editor to work.

**Comment by binyamin** [comment]
@juliushaertl After reading https://github.com/nextcloud/text/issues/323#issuecomment-538461645, I see that there are problems which plain-text editing would introduce.

Assuming that this feature is unlikely to be introduced in the near future, it would be helpful if the readme linked to https://github.com/nextcloud/text/issues/323#issuecomment-538461645 under a FAQ section. This issue has 42 comments, so it's difficult to process in a passing glance.

Also, I would consider renaming the app to **Rich**-Text. I didn't even know it _supported_ markdown syntax until I tried. Changing the name will likely clarify the editor's usages and capabilities. I think just renaming the "display name" (or whatever the equivalent is for nextcloud) would be sufficient; URLs and IDs should stay the same.

**Comment by rollanders** [comment]
I am not sure what is the problem. I guess it is simply a need for a FAQ.
I have 4 text editors installed, collabora, onlyoffice, Text and Plain text editor. 

I can create new txt-files by using Collabora (Document) or OnlyOffice (New document). In Collabora I need to change the default extension from odt to txt. txt-format is not present in the menu. In OnlyOffice I can choose txt format. 
When I clique on a txt-file it opens by default in the Text-app. It saves in the same format. There is no formating.
If I right clique on the txt-file I am given the possibility to edit the file in OnlyOffice and in Plain text editor. If i clique on the three dots I am also given the possibility to edit with Collabora. 

I can create new md-files by choosing New text document from the drop-down menu. If I clique on an md-file it opens in the Text app and I have the formatting menu. This editor mess up the md-files intended for cms_pico pages. 
If I rightclique on the md-file I can choose Edit in the plain text editor which has the option to show the formatted page side by side. I use this editor for the cms_pico files. 

I agree that the alternative in the drop-down menu could say New Rich text document for clarity. Maybe it wouls also be good to be able to use the Plain text editor from the drop down menue.

**Comment by E1k3** [comment]
I think it's ok that the Text app uses .md (markdown) files as long as it adheres to an accepted markdown definition and since markdown definitions are not exactly standardized (CommonMark is an attempt, but not universally adopted), different markdown editors frequently produce slightly different results.

What is definitely not ok, is to change an existing file just by opening it. Its only job is to conceal the existing markdown syntax and assist in formatting new content. At no point should it change the file without user interaction.

So instead of changing the file extension, I would propose fixing the Text app (i.e. stop changing files without user interaction) and add an option to change default mimetype <-> app associations.
This is done on all modern desktop environments and as long as Nextcloud allows apps to open certain files, it should also provide a mechanism to make these apps the default application for these file types.

**Comment by rollanders** [comment]
Something has changed. It seems that I now can edit the pico md-file with the  Text app without destroying the pico formatting. I am very impressed. Inserting pictures works OK. The formatting also. So the problem is solved as far as I can see.

**Comment by szaimen** [comment]
> Something has changed. It seems that I now can edit the pico md-file with the Text app without destroying the pico formatting. I am very impressed. Inserting pictures works OK. The formatting also. So the problem is solved as far as I can see.

Interesting! Which NC version are you using?

**Comment by rollanders** [comment]
I did this on my production server which has NC19.0.8, Pico CMS 1.0.12, Text 3.0.1

**Comment by rollanders** [comment]
But it is true there are changes in the YAML head of the PICO-md-file - the second line with --- was removed, and the second time I tested it I got a notice of a serverfault.  I then opened the md-file in the Simple text editor, removed ## in front of the first line in the YANL-header and readded the second line with --- and now the page was rendered OK. 

This is the page when edited in teh Text app
![Skärmklipp2_210520](https://user-images.githubusercontent.com/36486009/118935657-0f489480-b94c-11eb-92f5-854d13d44ed7.JPG)

This is what it looks like in the Simple text editor. The YAML header has been changed. This file gives server error
![Skärmklipp1_210520](https://user-images.githubusercontent.com/36486009/118935795-31daad80-b94c-11eb-8b75-499fc9cc5eb6.JPG)

I edit the YAML header in the Simple text editor 
![Skärmklipp5](https://user-images.githubusercontent.com/36486009/118935885-49199b00-b94c-11eb-938b-c10215025e93.JPG)

This renders OK 
![Skärmklipp4_210520](https://user-images.githubusercontent.com/36486009/118936055-77977600-b94c-11eb-8300-18b70fd8fec5.JPG)

**Comment by E1k3** [comment]
I suggest you open a new issue regarding Text changing files without interaction.

Side note: IMO having YAML inside markdown files is a questionable decision. Even more so when the [markdown definition](https://daringfireball.net/projects/markdown/syntax) picocms refers to in their documentation does not permit comments (where YAML could be stored without breaking markdown syntax).
